### PR TITLE
#320 Script storage option

### LINKED ISSUE #320: Add a single CLI option to compile for script storage
Aqua compiler provides the way to compile to pure AIR with `--air` option. And Fluence's script storage expects AIR to be stored there.

However, the produced AIR is wrong. The right set of options is `--no-xor --no-relay --air`, which is pretty confusing.

We need to have a single simple CLI option that sets all the internals to have the proper result for the script storage.

### PR DESCRIPTION
**Pull request by DieMyst**
fixes #320 

Adds `--script` flag to generate air for script storage

### CODE REVIEW

File: cli/src/main/scala/aqua/AppOpts.scala [hunk 1]
```diff
@@ -193,6 +193,12 @@ object AppOpts {
       .map(_ => true)
       .withDefault(false)
 
+  val scriptOpt: Opts[Boolean] =
+    Opts
+      .flag("script", "Generate code for script storage. Will ignore other options")
+      .map(_ => true)
```
Comment (alari) [inline cli/src/main/scala/aqua/AppOpts.scala:199]: .as(true)? Do you really need it?

File: cli/src/main/scala/aqua/AppOpts.scala [hunk 2]
```diff
@@ -193,6 +193,12 @@ object AppOpts {
       .map(_ => true)
       .withDefault(false)
 
+  val scriptOpt: Opts[Boolean] =
+    Opts
+      .flag("script", "Generate code for script storage. Will ignore other options")
```
Comment (alari) [inline cli/src/main/scala/aqua/AppOpts.scala:198]: Please elaborate what exactly it does

**Review by alari — APPROVED** [review]

File: cli/src/main/scala/aqua/AppOpts.scala (hunk 1 above)
Comment (DieMyst) [reply cli/src/main/scala/aqua/AppOpts.scala:199]: yes, cause it is unit otherwise

File: cli/src/main/scala/aqua/AquaCli.scala [hunk 3]
```diff
@@ -77,12 +78,13 @@ object AquaCli extends IOApp with Logging {
...
-            if (toAir) AirTarget
+            if (isScriptStorage) AirTarget
+            else if (toAir) AirTarget
             else if (toJs) JavaScriptTarget
             else TypescriptTarget
           val bc = {
-            val bc = TransformConfig(wrapWithXor = !noXor, constants = constants)
-            bc.copy(relayVarName = bc.relayVarName.filterNot(_ => noRelay))
+            val bc = TransformConfig(wrapWithXor = !(noXor || isScriptStorage), constants = constants)
+            bc.copy(relayVarName = bc.relayVarName.filterNot(_ => noRelay || isScriptStorage))
```
Comment (folex) [inline cli/src/main/scala/aqua/AquaCli.scala:87]: Maybe it's better to do like this:
```
if (isScriptStorage) {
  toAir = true
  noXor = true
  noRelay = true
}
```

This way, it's clear that `--script` is the same as `--no-xor --no-relay --air`

File: cli/src/main/scala/aqua/AppOpts.scala (hunk 2 above)
Comment (folex) [inline cli/src/main/scala/aqua/AppOpts.scala:198]: I think `--script` is kinda confusing. Everything you write in Aqua is a script.

I think `--scheduled` or maybe `--background` would be better.

```
.flag("scheduled", "To use as a scheduled script. Same as --air --no-relay --no-xor.")
```

**Review by folex — COMMENTED** [review]
Since `callbackSrv` is meaningless for scheduled scripts, I think it would be great to issue a warning or even an error if function returns a value is compiled under `--scheduled`/`--script`

**Pull request merged**